components-api/src/main/scala/pl/touk/nussknacker/engine/api/json/decoders/TypingResultDecoder.scala

```diff
@@ -42,6 +43,19 @@ class TypingResultDecoder(loadClass: String => Class[_]) {
     case e                     => Left(s"$e is not SingleTypingResult")
   }
 
+  private implicit val displayUnknownDecoder: Decoder[Unknown] = Decoder.instance { cursor =>
+    cursor
+      .as[String]
+      .map { displayStr =>
+        DisplayStrategy.valueOfDisplay(displayStr) match {
+          case Some(JsonDisplayStrategy)    => Typed.json
+          case Some(DefaultDisplayStrategy) => Unknown
+          case None =>
+            throw new IllegalStateException(s"Unrecognized display value: '$displayStr'")
```

Comment:
`IllegalStateException` should be reserved for cases when you put your application into a state that shouldn't happen. It is mostly used when you can't (or don't want to invest in it) express something in types and you have to handle something at runtime. The situation that you are handling here is a normal case - some client sent you unexpected data in a request. Circe prepared an error class for that. It is `DecodingFailure`. Also, it is more idiomatic to use `emap` or `.or(Decoder.failedWithMessage`.